rpc-versus/collect.py

```diff
@@ -0,0 +1,206 @@
+
+    def get_block_transaction_count_by_hash(self, block_hash):
+        return self.call_rpc("eth_getBlockTransactionCountByHash", [block_hash])
+
+    def get_uncle_count_by_number(self, block_number):
+        return self.call_rpc("eth_getUncleCountByBlockNumber", [hex(block_number)])
+
+    def get_uncle_count_by_hash(self, block_hash):
+        return self.call_rpc("eth_getUncleCountByBlockHash", [block_hash])
+
+    def get_block_receipts(self, block_number_or_hash):
+        return self.call_rpc("eth_getBlockReceipts", [block_number_or_hash])
+
+    # tx API
+
+    def get_transaction_by_hash(self, tx_hash): 
+        return self.call_rpc("eth_getTransactionByHash", [tx_hash])
+
+    def get_transaction_by_hash_index(self, block_hash, dex): 
+        return self.call_rpc("eth_getTransactionByBlockHashAndIndex", [block_hash, dex])
+
+    def get_transaction_by_number_index(self, block_number, dex): 
+        return self.call_rpc("eth_getTransactionByBlockNumberAndIndex", [hex(block_number), dex])
+
+    def get_transaction_receipt(self, tx_hash): 
+        return self.call_rpc("eth_getTransactionReceipt", [tx_hash])
+
+    def get_transaction_count(self, addr, block_number): 
+        return self.call_rpc("eth_getTransactionCount", [addr, hex(block_number)])
+
+    # logs API
+
+    def get_logs(self, filter_query):
+        return self.call_rpc("eth_getLogs", [filter_query])
+
+    # gas API
+
+    def fee_history(self, block_count, terminal_block, reward_percentiles):
+        return self.call_rpc("eth_feeHistory", [hex(block_count), hex(terminal_block), reward_percentiles])
+
+
+def aggregate_data(args):
+    endpoint = args.endpoint
+    file_name = args.filename
+    latest_block = int(args.latestblock)
+    number_of_blocks = args.blockrange
+
+    client = RPCClient(f'{endpoint}')
+        
+    results = {
+        'blocks': {'by_number':[],'by_hash':[],'tx_ct_by_num':[],'tx_ct_by_hsh':[],'ucl_ct_by_num':[],'ucl_ct_by_hsh':[]},
+        'txns': {'by_hash':[],'hash_dex':[],'num_dex':[],'receipt':[], 'counts':[]},
+        'receipts': [],
+        'logs': [],
+        'fees': []
+    }
+
+    for i, block_number in enumerate(range(latest_block, latest_block - number_of_blocks, -1)):
+
+        if block_number % 100 == 0:
+            print(f"inside collection loop, block: {block_number}")
+
+        prms = block_number
+        block_data = client.get_block_by_number(prms)
+        results['blocks']['by_number'].append({'arg':prms,'resp':block_data})
+        client.set_vars(block_data)
+
+        if len(client.tx_hashes) > 0:
+            prms = client.tx_hashes[0]
+            tx_data = client.get_transaction_by_hash(prms)
+            results['txns']['by_hash'].append({'arg':prms,'resp':tx_data})
+            prms = client.tx_hashes[-1]
+            tx_data = client.get_transaction_by_hash(prms)
+            results['txns']['by_hash'].append({'arg':prms,'resp':tx_data})
+
+        if i % 2 == 0:
```

Comment:
Why are we doing this every other block? This could use a comment at the very least.